### ISSUE TITLE
Compilation Issue: Unable to Find sawSensablePhantom Package on Ubuntu 18.04 ROS Melodic

### ISSUE THREAD

**Issue by xiancaiguazi** [jhu-saw/sawSensablePhantom#5]
### Issue Summary

I am working with the **SensAble PHANTOM Omni (FireWire/1394)** and attempting to compile the `sawSensablePhantom` package with minimal dependencies. I am following the instructions provided in the "Short version for Ubuntu (18.04) ROS (Melodic) to compile using catkin and wstool" section of `README.md`.

---

### Steps to Reproduce

1. I attempted to run the following command:
   ```bash
   wstool merge https://github.com/jhu-saw/sawSensablePhantom/raw/devel/ros/sensable_phantom.rosinstall
However, this resulted in an HTTP 404 error: Not Found.

2. To work around this, I manually downloaded the required files by checking the sensable_phantom.rosinstall file on the master branch. My current file structure is:
```
  - ct_ws
    - src
      - cisst-saw
          - cisst
          - cisstNetlib
          - cisst-ros
          - sawSensablePhantom
      - crtk
          - crtk_msgs
          - crtk_python_client
 ```

3. I proceeded with the build using `catkin build`. However, I encountered the following warning:
    ```bash
    Warnings   << sensable_phantom_ros:cmake /home/lab/ct_ws/logs/sensable_phantom_ros/build.cmake.003.log
    CMake Warning at /home/lab/ct_ws/src/cisst-saw/sawSensablePhantom/ros/CMakeLists.txt:58 (find_package):
      By not providing "FindsawSensablePhantom.cmake" in CMAKE_MODULE_PATH this
      project has asked CMake to find a package configuration file provided by
      "sawSensablePhantom", but CMake did not find one.
      Could not find a package configuration file provided by
      "sawSensablePhantom" (requested version 1.0.0) with any of the following
      names:
        sawSensablePhantomConfig.cmake
        sawsensablephantom-config.cmake
      Add the installation prefix of "sawSensablePhantom" to CMAKE_PREFIX_PATH or
      set "sawSensablePhantom_DIR" to a directory containing one of the above
      files.  If "sawSensablePhantom" provides a separate development package or
      SDK, be sure it has been installed.
    Information: code in /home/lab/ct_ws/src/cisst-saw/sawSensablePhantom/ros will not be compiled, it requires sawSensablePhantom

### Problem Description
The ROS node `sensable_phantom` is not compiled as a result of the above warning, and I am unable to run:
```
rosrun sensable_phantom_ros sensable_phantom
```

### Question
- How can I resolve the issue so that the sawSensablePhantom package is correctly found and built, allowing me to run the sensable_phantom_ros node successfully?
- Are there specific settings I need to configure in CMAKE_PREFIX_PATH or sawSensablePhantom_DIR to make this work?

**Comment by xiancaiguazi** [comment]
Thank you very much for reply.
I tried following solutions but still had issues.
1. Omni FireWire drivers has been installed.
I followed the [phantom-omni-1394-drivers](https://github.com/jhu-cisst-external/phantom-omni-1394-drivers) instruction and run [install-omni-drivers.sh](https://github.com/jhu-cisst-external/phantom-omni-1394-drivers/blob/master/install-omni-drivers.sh)
Once I ran `sudo /usr/sbin/PHANToMConfiguration`, I can see a serial number appear on the bottom right as said in [drivers.md](https://github.com/jhu-saw/sawSensablePhantom/blob/devel/drivers.md)

2. The device can be detected once plugged in.
As shown in the following command line output:
```
[573835.510813] firewire_core 0000:b3:00.0: created device fw1: GUID 000b990082565804, S400
[573835.510819] firewire_core 0000:b3:00.0: phy config: new root=ffc1, gap_count=5

```
3. `catkin build --force-cmake` still cannot get file complied, giving the same warning. By the way `catkin build --force-cmake-config` seems not a valid command.

4. Instead, I used the vcs to get new files, getting errors during compilation.
I downloaded the file using command
`vcs import --recursive --input https://raw.githubusercontent.com/jhu-saw/vcs/main/ros1 ros1-sensable-devel.vcs`
then set the compiling command to
`catkin config --cmake-args -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_BUILD_TYPE=Release`
However, the error showed during compilation:
```
/home/arclab/ciss_ws/src/cisst-saw/cisst/cisstNumerical/nmrNetlib.h:38:18: error: typedef redefinition with different types ('long' vs 'int')
typedef long int CISSTNETLIB_INTEGER;
                 ^
/home/arclab/autosurg_ws/devel/include/cisstNetlib/cisstNetlib-types.h:28:13: note: previous definition is here
typedef int CISSTNETLIB_INTEGER;
            ^
```
The compiler version is:
```
gcc (Ubuntu 9.4.0-1ubuntu1~18.04) 9.4.0
g++ (Ubuntu 9.4.0-1ubuntu1~18.04) 9.4.0
clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
```

**Comment by xiancaiguazi** [comment]
Hi, I wanted to report that the issue has been solved. I sucessfully built the main branch (i.e., the old version) using the files mentioned in sensable_phantom.rosinstall.

What I did was to follow the Omni FireWire drivers installation instruction again and recompliled the package. I guess [phantom-omni-1394-drivers](https://github.com/jhu-cisst-external/phantom-omni-1394-drivers) was not installed correctly before.

By the way, I also had the issue of " incompatible Qt library", which was solved by deactivating the conda and recompiling the package.
